src/filesystem/impls/filer/lib/HTMLRewriter.js

```diff
@@ -29,6 +29,15 @@ define(function (require, exports, module) {
             // Skip any links for protocols (we only want relative paths)
             var path = element.getAttribute(urlType);
             if(!Content.isRelativeURL(path)) {
+                if(urlType = 'script' && element.getAttribute('data-brackets-id') && $("#bramble-iframe-browser").hasClass("DoNotRunJavaScript")) {
```

Comment:
This needs to go into a new method, `HTMLRewriter.prototype.scripts` which will do special things for the `script` case.  I want to keep `HTMLRewriter.prototype.elements` clean for all other general cases.

Also, let's expose a public API function on this file in order to manage whether scripts are enabled/disabled (vs. using this class):

```js
var jsEnabled = true;
...
exports.enableScripts = function() {
  jsEnabled = true;
};
exports.disableScripts = function() {
  jsEnabled = false;
};
```

Now you can do this in your extension:

```js
var Rewriter = `brackets.getModule('filesystem/impls/filer/lib/HTMLRewriter');
...
// User clicked to stop scripts from running
Rewriter.disableScripts();
...
// User clicked to allow scripts to run again
Rewriter.enableScripts();
```

Then you can get rid of the extra class stuff you're doing, which is going to be fragile.